cpp/include/raft/neighbors/detail/nn_descent.cuh

```diff
@@ -1405,6 +1422,12 @@ void build(raft::resources const& res,
   auto int_graph = raft::make_host_matrix<int, int64_t, row_major>(
     dataset.extent(0), static_cast<int64_t>(extended_graph_degree));
 
+  auto distances_graph = raft::make_host_matrix<DistData_t, int64_t, row_major>(0, 0);
```

Comment:
Why is this extra allocation needed? Just copy from `h_dists` directly to the distance array allocated in the index.